0002/README.md

```diff
@@ -0,0 +1,257 @@
+ecosystem. When opted-in, package maintainers must upload provenance alongside
+the package proving that the package was built directly from the canonical
+source repository. At first this may just display results on a dashboard,
+similar to [pyreadiness.org](https://pyreadiness.org/3.10/), but will hopefully
+be integrated directly into PyPI eventually. With this milestone, software
+producers can opt-in to automatically protect themselves against insider risk,
+and software consumers can gain assurance for opted-in packages, but the package
+registry is still trusted.
+
+Design is P1 for [@var-sdk] and [@wietse-gmail] in Q2.
+
+Sub-problems:
+
+-   Policy model, which securely maps package name to canonical source
+    repository. For example, [requests](https://pypi.org/project/requests/) maps
+    to https://github.com/psf/requests. This metadata already exists for some
+    packages, but (a) it is not structured and (b) package maintainers can
+    mutate it at will. The solution must protect against malicious changes while
+    silently allowing official changes in source repository. One idea is to have
+    a Trust On First Use (TOFU) model with a mechanism for source repo
+    migrations.
+
+-   Implementation for each specific ecosystem, starting with Python/PyPI. This
+    includes a design, prototype, dashboard, Python community outreach, PEP
+    authoring and approval, and/or upload-time integration.
+
+-   (optional) Toy implementation of a
+    [mock package registry](https://github.com/jku/repository-playground) to
+    show how SLSA works in practice, without having to change any existing
+    system or to release any real package. This can also be used as examples for
+    https://slsa.dev/threats. (P3)
+
+#### Milestone: Client-side verification of SLSA policies in the Python ecosystem removes the need to trust the package registry (P3)
+
+SLSA verification happens in client tools that fetch Python packages, reducing
+or removing the need to trust the package registry (e.g. PyPI). This has a
+substantial reliability risk, so it is expected to come much later. Even in this
+model, the package registry enforces SLSA at upload time to prevent a maintainer
+from accidentally releasing a version that no one can use.
+
+This probably won't happen in 2022, but we list it here for completeness.
+
+### Theme: Adoption
+
+Widescale adoption of SLSA across open-source is the long-term goal, but it is
+not practical in 2022 because it not only depends on the work above, but also
+work to minimize or eliminate any friction for project maintainers.
+
+#### Milestone: Early adoption by key projects raises awareness of SLSA (P2)
+
+In the short term, we hope to have targeted adoption of key open-source projects
+to both exemplify standards and to reduce risk for those projects. We can also
+publish adoption stories to inform readers where SLSA is being applied. Finally,
```

Comment:
I really like this milestone- I see these adoption stories as great way to promote SLSA, show some of the techniques used to get there, and increase the security of the projects.